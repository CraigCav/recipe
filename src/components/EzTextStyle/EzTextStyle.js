import PropTypes from 'prop-types';
import styled from 'react-emotion';
import {variants} from '../../styles/';
import {standard} from '../../themes';
import {strong, subdued} from './styles';

const use = variants('use', {
  strong,
  subdued,
});

/**
 * Enhances text with styles to communicate emphasis.
 */
const EzTextStyle = styled('span')(use);

EzTextStyle.propTypes = {
  /**
   * The content that should be styled.
   */
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])),
  ]),
  /**
   * Describes the usage of the text in order to determine visual presentation.
   */
  use: PropTypes.oneOf(['strong', 'subdued']).isRequired,
};

/**
 * defaultProps
 * @property {object} theme - used the standard theme by default.
 */
EzTextStyle.defaultProps = {
  theme: standard,
};

/**
 * @component
 */
export default EzTextStyle;
