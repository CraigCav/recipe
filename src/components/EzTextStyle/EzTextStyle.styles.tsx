import {css} from 'react-emotion';

export const strong = ({theme}) => css`
  font-weight: ${theme.fontWeights.bold};
`;

export const subdued = ({theme}) => css`
  color: ${theme.colors.grays[600]};
`;
