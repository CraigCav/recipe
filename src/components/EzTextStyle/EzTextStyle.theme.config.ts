import {createCss} from '@stitches/core';
import root from '../theme.config';

export default createCss({
  ...root.config,
  theme: {
    fontWeights: {
      ...root.config.theme.fontWeights,
      'text-strong': '$fontWeights$bold',
    },
    colors: {
      ...root.config.theme.colors,
      'text-subdued': '$deemphasisText',
    },
  },
});
