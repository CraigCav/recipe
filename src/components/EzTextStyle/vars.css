:root {
  --recipe-text-strong-font-weight: var(--recipe-global-font-weight-bold);
  --recipe-text-subdued-text-color: var(--recipe-alias-deemphasis-text-color);
}