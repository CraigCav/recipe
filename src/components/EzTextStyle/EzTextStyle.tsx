import variant from 'styled-component-variant';
import {base, align, strong, subdued} from './EzTextStyle.styles';
import styled from '../../themes/styled';

const use = variant('use', {
  strong,
  subdued,
});

type Props = {
  use?: 'strong' | 'subdued';
  align?: 'left' | 'right' | 'center';
  children?: React.ReactNode;
};
/**
 * Enhances text with styles to communicate emphasis.
 */
const EzTextStyle = styled.span<Props>(base, use, align);

/**
 * @component
 */
export default EzTextStyle;
