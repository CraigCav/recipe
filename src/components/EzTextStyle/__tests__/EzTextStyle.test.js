import React from 'react';
import EzTextStyle from '../EzTextStyle';

describe('EzTextStyle', () => {
  /**
   * Style tests.
   */
  it('should render with strong styles', () => {
    const actual = create(<EzTextStyle use="strong">Strong Text</EzTextStyle>);
    expect(actual).toMatchSnapshot();
  });

  it('should render with subdued styles', () => {
    const actual = create(<EzTextStyle use="subdued">Subdued Text</EzTextStyle>);
    expect(actual).toMatchSnapshot();
  });

  it('renders as a span element', () => {
    const actual = shallow(<EzTextStyle use="strong">Strong Text</EzTextStyle>);
    expect(actual.name()).toEqual('span');
  });

  /**
   * Accessibility tests.
   */
  it('should meet accessibility guidelines', async () => {
    const wrapper = renderToHtml(<EzTextStyle use="strong">Strong Text</EzTextStyle>);
    const actual = await axe(wrapper);
    expect(actual).toHaveNoViolations();
  });
});
