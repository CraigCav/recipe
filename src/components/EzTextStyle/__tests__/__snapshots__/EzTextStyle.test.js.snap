// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EzTextStyle should render with strong styles 1`] = `
.recipe-0 {
  font-weight: 700;
}

<span
  className="recipe-0 recipe-1"
>
  Strong Text
</span>
`;

exports[`EzTextStyle should render with subdued styles 1`] = `
.recipe-0 {
  color: #8b99a6;
}

<span
  className="recipe-0 recipe-1"
>
  Subdued Text
</span>
`;
